src/server/game/Entities/Object/Object.h

```diff
@@ -489,8 +490,8 @@ class TC_GAME_API WorldObject : public Object, public WorldLocation
         // use only if you will sure about placing both object at same map
         bool IsWithinDist(WorldObject const* obj, float dist2compare, bool is3D = true) const;
         bool IsWithinDistInMap(WorldObject const* obj, float dist2compare, bool is3D = true) const;
-        bool IsWithinLOS(float x, float y, float z) const;
-        bool IsWithinLOSInMap(WorldObject const* obj) const;
+        bool IsWithinLOS(float x, float y, float z, ObjectIgnoreFlags ignoreFlags = ObjectIgnoreFlags::IGNORE_NONE) const;
```

Comment:
Enum scoping is required in case of enum class (see ObjectGuid.h for an example)